components/scream/src/physics/cld_fraction/atmosphere_cldfraction.cpp

```diff
@@ -0,0 +1,157 @@
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+  m_nk_pack  = ekat::npack<Spack>(m_num_levs);
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols,m_num_levs} };
+
+  // Set of fields used strictly as input
+  m_required_fields.emplace("qi",   scalar3d_layout_mid, Q,      grid_name);
+  m_required_fields.emplace("alst", scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used strictly as output
+  m_computed_fields.emplace("ast",   scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("aist",  scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used as input and output
+  // - There are no fields used as both input and output.
+
+}
+
+// =========================================================================================
+void CldFraction::initialize_impl (const util::TimeStamp& t0)
+{
+  m_current_ts = t0;
+}
+
+// =========================================================================================
+void CldFraction::run_impl (const Real dt)
+{
+/* Luca, do we need any of the following code?
+  // Copy inputs to host. Copy also outputs, cause we might "update" them, rather than overwrite them.
+  for (auto& it : m_cldfraction_fields_in) {
+    it.second.sync_to_host();
+  }
+  for (auto& it : m_cldfraction_fields_out) {
+    it.second.sync_to_host();
+  }
+
+  // Copy outputs back to device
+  // LB: why?!?
+  for (auto& it : m_cldfraction_fields_out) {
+    it.second.sync_to_dev();
+  }
+*/
+
+  
+  // Calculate ice cloud fraction and total cloud fraction given the liquid cloud fraction
+  // and the ice mass mixing ratio. 
+  auto qi   = m_cldfraction_fields_in["qi"].get_reshaped_view<const Pack**>();
+  auto alst = m_cldfraction_fields_in["alst"].get_reshaped_view<const Pack**>();
+  auto aist = m_cldfraction_fields_out["aist"].get_reshaped_view<Pack**>();
+  auto ast  = m_cldfraction_fields_out["ast"].get_reshaped_view<Pack**>();
+
+  cldfracF::cldfraction_main(m_num_cols,m_num_levs,qi,alst,aist,ast);
+
+  // Get a copy of the current timestamp (at the beginning of the step) and
+  // advance it,
+  auto ts = timestamp();
+  ts += dt;
+  for (auto& f : m_cldfraction_fields_out) {
+    f.second.get_header().get_tracking().update_time_stamp(ts);
+  }
+
+}
+
+// =========================================================================================
+void CldFraction::finalize_impl()
+{
+  // Do nothing
+}
+
+// =========================================================================================
+void CldFraction::register_fields (FieldRepository<Real>& field_repo) const {
+  std::set<ci_string> q_names =
+    { "qv","qc","qr","qi","qm","nc","nr","ni","bm"};
```

Comment:
most of these variables aren't used by cldfraction, so why register them here?

Reply:
Yes, I debated that myself.  I'm happy to drop it down to just `qi`, I left it in there this long just in case in the future we decide to use a scheme that relies on other tracer information.  But it is probably not unreasonable to expect the developer to also make this list longer if/when that happens.

Reply:
Yeah, let's not proactively add stuff people *might* want someday.